tests/app/broadcast_message/test_rest.py

```diff
@@ -145,20 +145,24 @@ def test_update_broadcast_message_allows_edit_while_not_yet_live(admin_request,
     t = create_template(sample_service, BROADCAST_TYPE)
     bm = create_broadcast_message(
         t,
-        areas={"areas": ['london'], "simple_polygons": [[50.12, 1.2], [50.13, 1.2], [50.14, 1.21]]},
+        areas={"areas": ['manchester'], "simple_polygons": [[50.12, 1.2], [50.13, 1.2], [50.14, 1.21]]},
         status=status
     )
 
     response = admin_request.post(
         'broadcast_message.update_broadcast_message',
-        _data={'starts_at': '2020-06-01 20:00:01', 'areas': ['london', 'glasgow']},
+        _data={
+            'starts_at': '2020-06-01 20:00:01',
+            'areas': ['london', 'glasgow'],
+            "simple_polygons": [[50.12, 1.2], [50.13, 1.2], [50.14, 1.21]]
+        },
```

Comment:
So good that you fixed this test but what I don't get was how was it passing before. Was it the lack of simple_polygons in the post request that was causing the update of the areas not to happen? If so, should that not have 400'd rather than 200'd?

Reply:
Hmm, I'm not sure, it's not a problem we would have in real life, more like bad test data. I could add a guard statement against that if you feel it would be better to, it would look something like:
```
if ('areas' in data and 'simple_polygons' not in data) or ('areas' not in data and 'simple_polygons' in data):
        raise InvalidRequest(
            f'Cannot update broadcast_message {broadcast_message.id}, areas or polygons are missing.',
            status_code=400
        )
```

Not sure what the benefit would be though.

Reply:
So I'd argue that it is worth testing for this and catching the error. The API is treated in theory as separate from the admin app so even though the admin app won't do this, we shouldn't trust that all API calls will follow this pattern. Maybe in the future the API endpoint will be accessible to the public. Maybe in the future a developer won't realise that you need to send both of them and just sends one of them. I think for the cost of the extra few lines of code if we always want both areas and polygons or neither, then it's worth us checking for this in the API.

I was trying to think of a smart way to do the if statement and the best I could come up with is
```if len(set(data).intersection(("areas", "simple_polygons"))) == 1```
but not sure how readable that is. Your suggestion is definitely more readable and so probably worth going for but it was a fun little brain teaser for me

Reply:
ok, I added that check ☺️ . Cool brain teaser with making the guard statement terse, code wars style :). As you say, I will keep the long-ish statement for readability ☺️